core/order.go

```diff
@@ -739,14 +809,35 @@ func validateCryptocurrencyOrderItem(item *pb.Order_Item) error {
 	return nil
 }
 
+func (n *OpenBazaarNode) getDivisibility(code string) uint32 {
+	dict := repo.LoadCurrencyDefinitions()
+	defn, err := dict.Lookup(code)
+	if err != nil {
+		return 0
+	}
+	return uint32(defn.Divisibility)
+}
+
+// GetCurrencyDefinition - return the currency defn for a coin
+func (n *OpenBazaarNode) GetCurrencyDefinition(code string) (*repo.CurrencyDefinition, error) {
+	dict := repo.LoadCurrencyDefinitions()
+	return dict.Lookup(code)
+}
+
 // EstimateOrderTotal - returns order total in satoshi/wei
-func (n *OpenBazaarNode) EstimateOrderTotal(data *PurchaseData) (uint64, error) {
+func (n *OpenBazaarNode) EstimateOrderTotal(data *PurchaseData) (big.Int, error) {
 	contract, err := n.createContractWithOrder(data)
 	if err != nil {
-		return 0, err
+		return *big.NewInt(0), err
 	}
 	payment := new(pb.Order_Payment)
-	payment.Coin = data.PaymentCoin
+	//payment.Coin = data.PaymentCoin
+	payment.AmountValue = &pb.CurrencyValue{
+		Currency: &pb.CurrencyDefinition{
+			Code:         data.PaymentCoin,
+			Divisibility: n.getDivisibility(data.PaymentCoin),
+		},
```

Comment:
I see that we're creating a lot of inline CurrencyDefinition protobufs. Would it better to do something like
```
d, err := repo.LoadCurrencyDefinitions.Lookup(data.PaymentCoin)
// ...so you can assign inside the CurrencyValue like
  Currency: d.Protobuf(),
```
since we're doing the lookup inside `getDivisibility` anyway? I would also add a unit test that every definition in the dictionary can successfully build it's own protobuf.